core/spi/src/main/java/org/jboss/arquillian/core/spi/Manager.java

```diff
@@ -46,4 +46,10 @@
 
     // clean
     void shutdown();
+
+    // adds extension specified via annotation @Observer
+    void addExtension(Class<?> extension) throws Exception;
+
+    // removes extension specified via annotation @Observer
```

Comment:
Not sure why we need these comments, I, for one, can clearly understand that from the method name. On top of that it might not only be used for `@Observer`, or at least nothing says here it only should be nor it's constrained to this concept.